borg/hashindex.pyx

```diff
@@ -212,7 +221,10 @@ cdef class ChunkIndex(IndexBase):
             raise KeyError(key)
         cdef int32_t refcount = _le32toh(data[0])
         if refcount != self.max_ref:
-            refcount -= 1
+            if self.max_ref - refcount < 1:
```

Comment:
it's unclear if this is enough. What's the internal python representation for max_ref when you do not specify type like that in the definition?

I think it's safer to do:
if refcount < 0 || self.max_ref - refcount < 1: